src/scalar.jl

```diff
@@ -50,90 +51,131 @@ struct Identity <: ScalarTransform end
 
-ScaledShiftedLogistic(scale::Real, shift::Real) =
-    ScaledShiftedLogistic(promote(scale, shift)...)
+inverse(t::TVScale, x::Number) = x / t.scale
+inverse_and_logjac(t::TVScale{<:Real}, x::Number) = inverse(t, x), -log(t.scale)
 
-# Switch to muladd and now it does have a DiffRule defined
-transform(t::ScaledShiftedLogistic, x::Real) = muladd(logistic(x), t.scale, t.shift)
+"""
+$(TYPEDEF)
 
-transform_and_logjac(t::ScaledShiftedLogistic, x) =
-    transform(t, x), log(t.scale) + logistic_logjac(x)
+Negative transformation `x ↦ -x`.
+"""
+struct TVNeg <: ScalarTransform
+end
+
+transform(::TVNeg, x::Real) = -x
+transform_and_logjac(t::TVNeg, x::Real) = transform(t, x), logjac_zero(LogJac(), typeof(x))
+
+inverse(::TVNeg, x::Number) = -x
+inverse_and_logjac(::TVNeg, x::Number) = -x, logjac_zero(LogJac(), typeof(x))
 
-function inverse(t::ScaledShiftedLogistic, y)
-    @argcheck y > t.shift           DomainError
-    @argcheck y < t.scale + t.shift DomainError
-    logit((y - t.shift)/t.scale)
+####
+#### composite scalar transforms
+####
+"""
+$(TYPEDEF)
+
+A composite scalar transformation, i.e. a sequence of scalar transformations.
+"""
+struct CompositeScalarTransform{Ts <: Tuple} <: ScalarTransform
+    transforms::Ts
+    function CompositeScalarTransform(transforms::Ts) where {Ts <: Tuple{ScalarTransform,Vararg{ScalarTransform}}}
+        new{Ts}(transforms)
+    end
 end
 
-# NOTE: inverse_and_logjac interface experimental and sporadically implemented for now
-function inverse_and_logjac(t::ScaledShiftedLogistic, y)
-    @argcheck y > t.shift           DomainError
-    @argcheck y < t.scale + t.shift DomainError
-    z = (y - t.shift) / t.scale
-    logit(z), logit_logjac(z) - log(t.scale)
+transform(t::CompositeScalarTransform, x) = foldr(transform, t.transforms, init=x)
+function transform_and_logjac(ts::CompositeScalarTransform, x) 
+    foldr(ts.transforms, init=(x, logjac_zero(LogJac(), typeof(x)))) do t, (x, logjac)
+        nx, nlogjac = transform_and_logjac(t, x)
+        (nx, logjac + nlogjac)
+    end
 end
 
+inverse(ts::CompositeScalarTransform, x) = foldl((y, t) -> inverse(t, y), ts.transforms, init=x)
+function inverse_and_logjac(ts::CompositeScalarTransform, x) 
+    foldl(ts.transforms, init=(x, logjac_zero(LogJac(), typeof(x)))) do (x, logjac), t
+        nx, nlogjac = inverse_and_logjac(t, x)
+        (nx, logjac + nlogjac)
+    end
+end
+
+Base.:∘(t::ScalarTransform, s::ScalarTransform) = CompositeScalarTransform((t, s))
+Base.:∘(t::ScalarTransform, ct::CompositeScalarTransform) = CompositeScalarTransform((t, ct.transforms...))
+Base.:∘(ct::CompositeScalarTransform, t::ScalarTransform) = CompositeScalarTransform((ct.transforms..., t))
+Base.:∘(ct1::CompositeScalarTransform, ct2::CompositeScalarTransform) = CompositeScalarTransform((ct1.transforms..., ct2.transforms...))
+Base.:∘(t::ScalarTransform, tt::Vararg{ScalarTransform}) = CompositeScalarTransform((t, tt...))
```

Comment:
Is this actually needed? It also seems wrong if a few of `tt` are `CompositeScalarTransform`. Based on how this expression is parsed, I assumed it's sufficient to define `∘` for two arguments:

```julia
julia> Meta.dump(:(f ∘ g ∘ h))
Expr
  head: Symbol call
  args: Array{Any}((3,))
    1: Symbol ∘
    2: Expr
      head: Symbol call
      args: Array{Any}((3,))
        1: Symbol ∘
        2: Symbol f
        3: Symbol g
    3: Symbol h
```

Reply:
```suggestion
```

Reply:
I like the idea of allowing `compose(t1, t2, t3, t4)`, rather than forcing `compose(compose(compose(t1, t2), t3), t4)`, so that was my intention for this method. But you are right that it can give somewhat wrong results if there are `CompositeScalarTransform`s inside the `Vararg`--fixed this now with a `foldl` that will recursively call the two-argument forms, and added tests that catch this case.